pkg/terminal/proxy.go

```diff
@@ -113,22 +108,22 @@ func (p *Proxy) HandleProxy(user *auth.User, w http.ResponseWriter, r *http.Requ
 		return
 	}
 
-	client, err := p.createDynamicClient(user.Token)
-	if err != nil {
-		http.Error(w, "Failed to create k8s client for the authenticated user. Cause: "+err.Error(), http.StatusInternalServerError)
-		return
-	}
-
 	userId := user.ID
 	if userId == "" {
+		client, err := p.createTypedClient(user.Token)
+		if err != nil {
+			http.Error(w, "Failed to create k8s client for the authenticated user. Cause: "+err.Error(), http.StatusInternalServerError)
+			return
+		}
+
 		// user id is missing, auth is used that does not support user info propagated, like OpenShift OAuth
-		userInfo, err := client.Resource(UserGroupVersionResource).Get(context.TODO(), "~", metav1.GetOptions{})
+		userInfo, err := client.AuthenticationV1().SelfSubjectReviews().Create(r.Context(), &v1.SelfSubjectReview{}, metav1.CreateOptions{})
 		if err != nil {
 			http.Error(w, "Failed to retrieve the current user info. Cause: "+err.Error(), http.StatusInternalServerError)
 			return
 		}
 
-		userId = string(userInfo.GetUID())
+		userId = string(userInfo.Status.UserInfo.UID) // TODO: are we cool rewriting this here?
 		if userId == "" {
 			// uid is missing. it must be kube:admin
 			if "kube:admin" != userInfo.GetName() {
```

Comment:
Is `userInfo.getName()` giving us the name of the user or the name of the SSR instance?